src/sentry/api/serializers/snuba.py

```diff
@@ -150,7 +150,7 @@ def zerofill(data, start, end, rollup):
     # Add any remaining rows that are not aligned to the rollup and are lower than the
     # end date.
     if i < len(data):
-        rv.extend(row for row in data[i:] if row[0] < rollup_end)
+        rv.extend(row for row in data[i:] if row[0] < end)
```

Comment:
Take the test below for example. If `start` is 9:30 and `end` is 11:30 with `rollup` of 1 hour. This will cause `rollup_start` to be 9:00 and `rollup_end` to be 11:00.

This means that if we have any data that satisfies 11:00 < timestamp < 11:30, it will be lost here despite it satisfying 9:30 < timestamp < 11:30